RiotSwiftUI/Modules/Room/PollHistory/Service/MatrixSDK/PollHistoryService.swift

```diff
@@ -112,21 +158,41 @@ private extension PollHistoryService {
         currentBatchSubject = nil
     }
     
-    func aggregatePoll(pollStartEvent: MXEvent) {
-        guard pollAggregators[pollStartEvent.eventId] == nil else {
-            return
-        }
+    func aggregatePoll(pollStartEvent: MXEvent, isLivePoll: Bool) {
+        let eventId: String = pollStartEvent.eventId
         
-        guard let aggregator = try? PollAggregator(session: room.mxSession, room: room, pollEvent: pollStartEvent, delegate: self) else {
+        guard pollAggregationContexts[eventId] == nil else {
             return
         }
         
-        pollAggregators[pollStartEvent.eventId] = aggregator
+        let newContext: PollAggregationContext = .init(isLivePoll: isLivePoll)
+        pollAggregationContexts[eventId] = newContext
+        
+        do {
+            newContext.pollAggregator = try PollAggregator(session: room.mxSession, room: room, pollEvent: pollStartEvent, delegate: self)
+        } catch {
+            pollAggregationContexts.removeValue(forKey: eventId)
+        }
     }
     
     var timestampTargetReached: Bool {
         oldestEventDate <= targetTimestamp
     }
+    
+    var oldestEventDate: Date {
+        get {
+            oldestEventDateSubject.value
+        }
+        set {
+            oldestEventDateSubject.send(newValue)
+        }
+    }
+}
+
+private extension Date {
+    func subtractingDays(_ days: UInt) -> Date {
+        addingTimeInterval(-TimeInterval(days) * PollHistoryConstants.oneDayInSeconds)
```

Comment:
And again with the calendar something like so:
`Calendar.current.date(byAdding: DateComponents(day: -days), to: self)`

Reply:
Applied even if it has a downside: it returns an optional Date. :-/
https://github.com/vector-im/element-ios/pull/7303/commits/fc8ee97c1b106c20b75cecfebef73c35e036dbec